### ISSUE TITLE
Missing definitions for BUILTIN_STL_SUPPORT

### ISSUE THREAD

**Issue by albert-github** [doxygen/doxygen#6996]
When we have the following source code:

The `.cc` file:
```
using namespace std;

theClass::theClass(const vector<string>& fileNames, const std::vector<std::pair<std::string, std::string>>& members) {}
theClass::theClass(const string& grbmlFileName, const std::vector<std::pair<std::string, std::string>>& members) {}
```

The `.h` file:
```
class theClass
{
public:
    theClass(const std::vector<std::string>& fileNames, const std::vector<std::pair<std::string, std::string>>& members);
    theClass(const std::string& grbmlFileName, const std::vector<std::pair<std::string, std::string>>& members);
};
```

We get warnings like:
```
.../tst.cc:3: warning: no matching class member found for
  theClass::theClass(const std::vector< std::string > &fileNames, const std::vector< std::pair< std::std::string, std::std::string >> &members)
Possible candidates:
  theClass::theClass(const std::vector< std::string > &fileNames, const std::vector< std::pair< std::string, std::string >> &members)
  theClass::theClass(const std::string &grbmlFileName, const std::vector< std::pair< std::string, std::string >> &members)

.../tst.cc:4: warning: no matching class member found for
  theClass::theClass(const std::string &grbmlFileName, const std::vector< std::pair< std::std::string, std::std::string >> &members)
Possible candidates:
  theClass::theClass(const std::vector< std::string > &fileNames, const std::vector< std::pair< std::string, std::string >> &members)
  theClass::theClass(const std::string &grbmlFileName, const std::vector< std::pair< std::string, std::string >> &members)
```

The problem is caused by the fact that `std::pair` is not recognized in the `BUILTIN_STL_SUPPORT`, so we probably need a line like:
```
{ "pair",        0,                              0,                     "T",           "elements",        "T",          "elements",    FALSE,              TRUE  }, // C++11
```
in `doxygen.cpp` (though I'm not sure about the different fields here). Same accounts for `tuple` (and probably some others).

The case: [example.tar.gz](https://github.com/doxygen/doxygen/files/13648174/example.tar.gz)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
